src/main/java/com/google/api/codegen/discogapic/transformer/DiscoGapicParser.java

```diff
@@ -142,20 +128,34 @@ public static Name methodAsName(Method method) {
 
   /**
    * Return the name of the fully qualified resource from a given canonicalized path. Use {@link
-   * #getCanonicalPath(Method)}} for canonicalization of the parameter.
+   * #getCanonicalPath(String)}} for canonicalization of the parameter. This method includes all
+   * wildcards from the input path in the resulting Name, and it includes literal segments from the
+   * input path
    */
   public static Name getQualifiedResourceIdentifier(String canonicalPath) {
     String[] pieces = canonicalPath.split(PATH_DELIMITER);
 
     Name name = null;
-    for (String piece : pieces) {
-      if (!piece.contains("{")) {
-        if (name == null) {
-          name = Name.from(Inflector.singularize(piece));
-        } else {
-          name = name.join(stringToName(Inflector.singularize(piece)));
+    for (int i = 0; i < pieces.length; i++) {
+      if (i < pieces.length - 1 && pieces[i + 1].contains("{")) {
+        // Current index at a literal segment directly preceding a wildcard segment.
+        // Skip literal segment iff it is the same as or the pluralized version of the wildcard string.
+        String wildCard = pieces[i + 1].substring(1, pieces[i + 1].length() - 1);
+        if (pieces[i].toLowerCase().equals(wildCard.toLowerCase())
+            || Inflector.singularize(pieces[i].toLowerCase()).equals(wildCard.toLowerCase())) {
+          continue;
         }
       }
+
+      String namePiece = pieces[i];
+      if (namePiece.contains("}")) {
+        namePiece = namePiece.substring(1, namePiece.length() - 1);
+      }
+      if (name == null) {
```

Comment:
Let's initialize `name` with `Name name = Name.from()`, then we can remove this if statement.

Reply:
Sure.